synapse/handlers/auth.py

```diff
@@ -617,6 +617,17 @@ def add_threepid(self, user_id, medium, address, validated_at):
             self.hs.get_clock().time_msec()
         )
 
+    @defer.inlineCallbacks
+    def delete_threepid(self, user_id, medium, address):
+        # 'Canonicalise' email addresses as per above
```

Comment:
you're going to hate me for asking this, but: should we be canonicalising MSISDNs here? or are we assuming the client has already determined normal form?  (If so, a comment would be useful)